src/main/java/com/epam/task/purchases/Euro.java

```diff
@@ -0,0 +1,58 @@
+package com.epam.task.purchases;
+
+public class Euro implements Comparable<Euro>{
+    private final int value;
+
+    public Euro(int value) {
+        this.value = value;
+    }
+
+    public Euro(int euros, int cents){
+        this.value = euros*100 + cents;
+    }
+
+    public Euro(Euro euro){
+        this.value = euro.value;
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    public Euro add(Euro euro){
+        return new Euro(this.value + euro.value);
+    }
+
+    public Euro sub(Euro euro){
+        return new Euro(this.value - euro.value);
+    }
+
+    public Euro mul(int k){
+        return new Euro(this.value * k);
+    }
+
+    public Euro mul(double x, RoundMethods roundMethod, int d){
+        return new Euro(roundMethod.round(value * x, roundMethod, d));
```

Comment:
```suggestion
        return new Euro(roundMethod.round(value * x, d));
```
try to implement it like this.